src/ir.cpp

```diff
@@ -23608,6 +23615,24 @@ static Error ir_make_type_info_decls(IrAnalyze *ira, IrInst* source_instr, ZigVa
     ensure_field_index(type_info_declaration_type, "is_pub", 1);
     ensure_field_index(type_info_declaration_type, "data", 2);
 
+    if (!resolve_types) {
+        ZigType *ptr_type = get_pointer_to_type_extra(ira->codegen, type_info_declaration_type,
+            false, false, PtrLenUnknown, 0, 0, 0, false);
+
+        out_val->special = ConstValSpecialLazy;
+        out_val->type = get_slice_type(ira->codegen, ptr_type);
+
+        LazyValueTypeInfoDecls *lazy_type_info_decls = heap::c_allocator.create<LazyValueTypeInfoDecls>();
```

Comment:
This makes 1 lazy value for the entire thing, which means that if you access one decl, it brings in everything. I think this would be better to make a lazy value for each decl. 

Or maybe this is still good and it can be a multi-layered lazy thing, and in resolving LazyValueIdTypeInfoDecls it would create lazy values for each element.

I think this is related to #4495.

It's the end of a long 2 days for me, I'm going to come back to this tomorrow, and we'll get this merged :+1:

Reply:
Yes, I did that intentionally, because it seemed easier to implement at that point. However, I can now try to make each element a lazy value, will do that tonight after work :)

One thing I'm wondering is, what should happen when the slice is used in a for loop. Should it then resolve every value in the range of the slice?